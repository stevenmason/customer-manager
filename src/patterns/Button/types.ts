interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  href?: string;
}

export default Props;
