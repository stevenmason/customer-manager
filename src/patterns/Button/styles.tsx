import styled from "styled-components";

const Button = styled.button`
  margin: 10px;
  background-color: white;
  border: 1px solid #ced4da;
  color: #5d6369;
  font-size: 1.5rem;
  border-radius: 0.25rem;
  padding: 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  cursor: pointer;
`;

export { Button };
