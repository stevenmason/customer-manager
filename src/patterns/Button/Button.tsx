import React from "react";

import Props from "./types";
import * as SC from "./styles";

const openLink = (href: string) => window.open(href, "_blank");

const Button: React.FC<Props> = ({ children, href, onClick }) => (
  <SC.Button onClick={href ? () => openLink(href) : onClick}>
    {children}
  </SC.Button>
);

export default Button;
