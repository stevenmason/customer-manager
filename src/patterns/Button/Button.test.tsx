import React from "react";
import renderer from "react-test-renderer";

import Button from "./Button";

const PROPS = { onClick: () => {} };

it("renders correctly", () => {
  const tree = renderer.create(<Button {...PROPS} />).toJSON();
  expect(tree).toMatchSnapshot();
});
