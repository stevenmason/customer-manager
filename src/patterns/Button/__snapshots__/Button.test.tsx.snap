// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="sc-bdVaJa eneQrO"
  onClick={[Function]}
/>
`;
